src/connectors/h1_collect.py

```diff
@@ -0,0 +1,316 @@
+    },
+    {
+        'name': 'api_identifier',
+        'title': "API Identifier",
+        'prompt': "Available in your H1 Settings",
+        'type': 'str',
+        'secret': True,
+        'required': True,
+    },
+    {
+        'name': 'program_name',
+        'title': "Program Name",
+        'prompt': "Your handle in Program Settings",
+        'type': 'str',
+        'required': True,
+    },
+    {
+        'name': 'account_id',
+        'title': "Account ID",
+        'prompt': "https://api.hackerone.com/core-resources/#programs-get-your-programs",
+        'type': 'str',
+        'required': False,
+    },
+]
+
+LANDING_TABLE_COLUMNS_TRANSACTIONS = [
+    ('timestamp', 'TIMESTAMP_LTZ(9)'),
+    ('type', 'string'),
+    ('activity_date', 'TIMESTAMP_LTZ(9)'),
+    ('activity_description', 'string'),
+    ('bounty_award', 'int'),
+    ('bounty_fee', 'int'),
+    ('debit_or_credit_amount', 'int'),
+    ('balance', 'int'),
+    ('id', 'number'),
+    ('url', 'string'),
+]
+
+LANDING_TABLE_COLUMNS_REPORTS = [
+    ('timestamp', 'TIMESTAMP_LTZ(9)'),
+    ('id', 'number'),
+    ('type', 'string'),
+    ('title', 'string'),
+    ('state', 'string'),
+    ('created_at', 'TIMESTAMP_LTZ(9)'),
+    ('vulnerability_information', 'string'),
+    ('triaged_at', 'TIMESTAMP_LTZ(9)'),
+    ('closed_at', 'TIMESTAMP_LTZ(9)'),
+    ('first_program_activity_at', 'TIMESTAMP_LTZ(9)'),
+    ('bounty_awarded_at', 'TIMESTAMP_LTZ(9)'),
+    ('rating', 'string'),
+    ('name', 'string'),
+    ('description', 'string'),
+    ('external_id', 'string'),
+    ('asset_identifier', 'string'),
+    ('awarded_amount', 'int'),
+    ('awarded_bonus_amount', 'int'),
+]
+
+
+# Perform the API call
+def load_data(url: str, token: str, api_identifier: str, params: dict = {}) -> dict:
+    headers: dict = {"Accept": "application/json"}
+
+    try:
+        log.debug(f"Preparing GET: url={url} with params={params}")
+        resp = requests.get(
+            url, auth=(api_identifier, token), params=params, headers=headers
+        )
+        resp.raise_for_status()
+    except HTTPError as http_err:
+        log.error(f"Error GET: url={url}")
+        log.error(f"HTTP error occurred: {http_err}")
+        raise
+    log.debug(resp.status_code)
+    return resp.json()['data']
+
+
+def get_path(data, path, default=None):
+    for key in path.split("."):
+        if key not in data:
+            return default
+        else:
+            data = data[key]
+    return data
+
+
+def connect(connection_name, options):
+    landing_transactions_table = f'data.h1_collect_transactions_connection'
+    landing_reports_table = f'data.h1_collect_reports_connection'
+
+    comment = yaml_dump(module='h1_collect', **options)
+
+    db.create_table(
+        name=landing_transactions_table,
+        cols=LANDING_TABLE_COLUMNS_TRANSACTIONS,
+        comment=comment,
+    )
+    db.execute(
+        f'GRANT INSERT, SELECT ON {landing_transactions_table} TO ROLE {SA_ROLE}'
```

Comment:
iirc, this can be passed as `rw_role` to `create_table`

Reply:
Do we remove lines 122-124 and replace it with `rw_role=SA_ROLE` in the `create_table` call?

Reply:
Yes, like [here](https://github.com/snowflakedb/SnowAlert/blob/40f38a3f7fbd9c3cde7e7e51dd72ae881a70cd0c/src/connectors/airwatch_devices.py#L148-L153).